tests/data/MerchFactory.ts

```diff
@@ -38,22 +53,37 @@ export class MerchFactory {
     return substitutes.map((sub) => MerchandiseItemOptionModel.merge(MerchFactory.fakeOption(), sub));
   }
 
+  public static optionMetadataWith(...substitutes: Partial<MerchItemOptionMetadata>[]): MerchItemOptionMetadata[] {
+    return substitutes.map((sub) => {
+      const metadata = MerchFactory.fakeOptionMetadata();
+      return {
+        type: sub.type ?? metadata.type,
+        value: sub.value ?? metadata.value,
+        position: sub.position ?? metadata.position,
+      };
```

Comment:
```suggestion
      return { ...metadata, ...sub };
```
This should do the same thing I think.